R/colocboost_pipline.R

```diff
@@ -0,0 +1,667 @@
+                             maf_cutoff = 0.0005, 
+                             pip_cutoff_to_skip_ind = 0,
+                             # - sumstat
+                             skip_region = NULL,
+                             remove_indels = FALSE,
+                             pip_cutoff_to_skip_sumstat = 0,
+                             qc_method = c("rss_qc", "dentist", "slalom"),
+                             impute = TRUE, 
+                             impute_opts = list(rcond = 0.01, R2_threshold = 0.6, minimum_ld = 5, lamb = 0.01)){
+    
+    # - individual level data QC
+    individual_data <- region_data$individual_data
+    if (!is.null(individual_data)){
+        X <- individual_data$residual_X
+        Y <- individual_data$residual_Y
+        MAF <- individual_data$maf
+        # 1. remove maf < maf_cutoff
+        # 2. initial check PIP
+        individual_data <- data_initial_screen_individual(X = X, Y = Y, MAF = MAF, 
+                                                          maf_cutoff = maf_cutoff, 
+                                                          pip_cutoff_to_skip = pip_cutoff_to_skip_ind )
+    }
+    
+    # - summary statistics QC
+    sumstat_data = region_data$sumstat_data
+    if (!is.null(sumstat_data)){
+        # - 1. initial check PIP
+        sumstat_data <- data_initial_screen_sumstat(sumstat_data, skip_region = skip_region,
+                                                    remove_indels = remove_indels, 
+                                                    pip_cutoff_to_skip = pip_cutoff_to_skip_sumstat)
+        
+        # - qc or impute
+        sumstat_data <- sumstat_qc(sumstat_data, qc_method = qc_method,
+                                   impute = impute, impute_opts = impute_opts)
+        
+    }
+    return(list(individual_data = individual_data, 
+                sumstat_data = sumstat_data))
+
+}
+
+
+
+
+
+############# Individual - level data
+#' Add context names to colname of Y if missing
+#' @return A list of residual Y
+#' @noRd
+add_context_to_Y <- function(res_Y){
+    res <- lapply(1:length(res_Y), function(iy){
+        y <- res_Y[[iy]]
+        if (is.null(colnames(y))){ 
+            colnames(y) <- names(res_Y)[iy] 
+        } else {
+            colnames(y) <- paste0(names(res_Y)[iy], "_", colnames(y))
+        }
+        return(y)
+    })
+    names(res) <- names(res_Y)
+    return(res)
+}
+
+#' Remove variants with maf < maf_cutoff
+#' @return A list of residual X and maf
+#' @export
+filter_resX_maf <- function(res_X, maf, maf_cutoff = 0.0005){
+    pos <- which(maf < maf_cutoff)
+    if (length(pos)!=0){
+        res_X <- res_X[, -pos, drop = FALSE] %>% .[, order(colnames(.)), drop = FALSE]
+        maf <- maf[-pos]
+    }
+    return(list(res_X = res_X, maf = maf))
+}
+
+#' Initial PIP check for individual level data
+#' @return A list of residual Y
+#' @export
+filter_resY_pip <- function(res_X, res_Y, pip_cutoff_to_skip = 0, context = NULL){
```

Comment:
We also have something similar here: https://github.com/StatFunGen/pecotmr/blob/main/R/multivariate_pipeline.R#L99 if you think it is cleaner to keep using your customized version I would suggest hiding this function into the pipeline function itself (just like this `skip_conditions` function here) not making it an explicit function that gets exported.